ast-nodes/expressions/index-access.dart

```diff
@@ -45,6 +47,13 @@ class IndexAccess implements ModifiablePrimary {
   }
 
   void checkSemantics() {
-    // TODO: implement
+    this.object.checkSemantics();
+    this.index.checkSemantics();
+
+    if (this.object.resultType is! ArrayType) {
```

Comment:
Shouldn't we also check, if index expression is of integer type? And if it is in range

Reply:
The type, yes, you're right. As for the range, this information is not always known at compile time, so this will have to be checked at runtime